src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,126 @@
+
+    private Comparator<Person> personByLastAndFirstNamesComparator = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            int lastNameCompare = o1.getLastName().compareTo(o2.getLastName());
+            if (lastNameCompare != 0) {
+                return lastNameCompare;
+            }
+            return o1.getFirstName().compareTo(o2.getFirstName());
+        }
+    };
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons,personComparator);
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, personByLastAndFirstNamesComparator);
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Person person = FluentIterable
+            .from(persons)
+            .firstMatch(new Predicate<Person>() {
+                @Override
+                public boolean apply(Person person) {
+                    return person.getAge() == 30;
+                }
+            })
+            .get();
+
+        assertEquals(new Person("Николай", "Зимов", 30), person);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaAnonymousPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Predicate<Person> personPredicate = new Predicate<Person>() {
+            @Override
+            public boolean apply(Person person) {
+                return person.getAge() == 30;
+            }
+        };
+
+        Person person = FluentIterable
+            .from(persons)
+            .firstMatch(personPredicate)
+            .get();
```

Comment:
It's better to use ```orNull()``` or ```or(defaultValue)``` if you don't like to catch exceptions like ```NoSuchElementException```.

Reply:
Yep, fixed